statistics/table.go

```diff
@@ -155,74 +170,87 @@ func (t *Table) String() string {
 	return strings.Join(strs, "\n")
 }
 
-// ColumnIsInvalid checks if this column is invalid.
-func (t *Table) ColumnIsInvalid(colInfo *model.ColumnInfo) bool {
+type neededColumnMap struct {
+	m    sync.Mutex
+	cols map[int64][]*Column
+}
+
+var neededColumns = neededColumnMap{cols: map[int64][]*Column{}}
+
+// ColumnIsInvalid checks if this column is invalid. If this column has histogram but not loaded yet, then we mark it
+// as need histogram.
+func (t *Table) ColumnIsInvalid(colID int64) bool {
```

Comment:
I think it would be useful for the call to know that if the column histogram is not loaded.
For example, the plan cache should not cache the plan if the histogram is not loaded, or if the cost of the plan is very big, we can actively load the histogram before execution.

We can store this information in `StatementContext`.